### PR TITLE
Refactor golang packages

### PR DESCRIPTION
**Pull request by gmelodie**
Broke the code down into different packages using `go mod`. This should solve linting errors :smiley:

### CODE REVIEW

**Comment by IamCathal** [comment]
Queries are going to have to be overwtitten by the new ones introduced in  https://github.com/MLH-Fellowship/FellowshipWrapup/pull/21

There are some API changes that I need to make but that's better left for after this has been merged to master. Apart from that it's fine after the conflicts have been resolved or just ignored since this is an architecture change

**Comment by gmelodie** [comment]
This is ready. Conflicts are resolved but I'm still getting `Variable $username is used by  but not declared`.

File: backend/util/util.go [hunk 1]
```diff
@@ -127,19 +125,20 @@ func IsValidUsername(username string) bool {
...
-		return false
+		// couldn't decode the POST data into JSON
+		return false, err
 	}
-	if req.Secret == os.Getenv("secretKey") {
-		return true
+
+	if req.Secret != os.Getenv("secretKey") {
+		fmt.Println(req.Secret)
+		return false, errors.New("Incorrect 'secret'")
```
Comment (IamCathal) [inline backend/util/util.go:145]: I think "Incorrect secretKey environvment variable" would be more descriptive if the user didn't know that a `secretKey` needed to set

File: backend/server/server.go [hunk 2]
```diff
@@ -19,9 +17,55 @@ type response struct {
...
+		if vars["username"] == "" {
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusBadRequest)
+			res := response{
+				Status: "error",
+				Body:   "No username given",
+			}
+			json.NewEncoder(w).Encode(res)
+
+			util.LogCall("POST", r.RequestURI, "400")
```
Comment (IamCathal) [inline backend/server/server.go]:
```suggestion
			util.LogCall(r.Method, r.RequestURI, "400")
```

File: backend/server/server.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		if v, err := util.IsValidUsername(vars["username"]); !v {
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusBadRequest)
+			res := response{
+				Status: "422",
+				Body:   fmt.Sprint(err),
+			}
+			json.NewEncoder(w).Encode(res)
+
+			util.LogCall("POST", r.RequestURI, "400")
```
Comment (IamCathal) [inline backend/server/server.go:46]:
```suggestion
			util.LogCall(r.Method, r.RequestURI, "400")
```

File: backend/util/util.go (hunk 1 above)
Comment (gmelodie) [reply backend/util/util.go:145]: Didn't implement because I'm cooler.

**Pull request merged**